flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RelDataTypeJsonSerializer.java

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.serde;
+
+import org.apache.flink.table.planner.plan.schema.GenericRelDataType;
+import org.apache.flink.table.planner.plan.schema.RawRelDataType;
+import org.apache.flink.table.planner.plan.schema.StructuredRelDataType;
+import org.apache.flink.table.planner.plan.schema.TimeIndicatorRelDataType;
+import org.apache.flink.table.types.logical.TimestampKind;
+import org.apache.flink.table.types.logical.TypeInformationRawType;
+import org.apache.flink.table.utils.EncodingUtils;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.JsonGenerator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.SerializerProvider;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.MapSqlType;
+import org.apache.calcite.sql.type.MultisetSqlType;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import java.io.IOException;
+
+/**
+ * JSON serializer for {@link RelDataType}. refer to {@link RelDataTypeJsonDeserializer} for
+ * deserializer.
+ */
+public class RelDataTypeJsonSerializer extends StdSerializer<RelDataType> {
+    private static final long serialVersionUID = 1L;
+
+    public static final String FIELD_NAME_TYPE_NAME = "typeName";
+    public static final String FIELD_NAME_FILED_NAME = "fieldName";
+    public static final String FIELD_NAME_NULLABLE = "nullable";
+    public static final String FIELD_NAME_PRECISION = "precision";
+    public static final String FIELD_NAME_SCALE = "scale";
+    public static final String FIELD_NAME_FIELDS = "fields";
+    public static final String FIELD_NAME_STRUCT_KIND = "structKind";
+    public static final String FIELD_NAME_TIMESTAMP_KIND = "timestampKind";
+    public static final String FIELD_NAME_ELEMENT = "element";
+    public static final String FIELD_NAME_KEY = "key";
+    public static final String FIELD_NAME_VALUE = "value";
+    public static final String FIELD_NAME_TYPE_INFO = "typeInfo";
+    public static final String FIELD_NAME_RAW_TYPE = "rawType";
+    public static final String FIELD_NAME_STRUCTURED_TYPE = "structuredType";
+
+    public RelDataTypeJsonSerializer() {
+        super(RelDataType.class);
+    }
+
+    @Override
+    public void serialize(
+            RelDataType relDataType,
+            JsonGenerator jsonGenerator,
+            SerializerProvider serializerProvider)
+            throws IOException {
+        jsonGenerator.writeStartObject();
+        serialize(relDataType, jsonGenerator);
+        jsonGenerator.writeEndObject();
+    }
+
+    private void serialize(RelDataType relDataType, JsonGenerator gen) throws IOException {
+        if (relDataType instanceof TimeIndicatorRelDataType) {
```

Comment:
can we reuse the Serde of LogicalType so that we don't need to have two kinds of type in json? such as FlinkTypeFactory#toLogicalType, I think Logic Type may be eable to cover RelDataType we are interested in.

Reply:
Some RelDataTypes are not support, such as `ANY`, `TIME_WITH_LOCAL_TIME_ZONE`, `SYMBOL`, `REAL`, `DISTINCT`